### DISCUSSION TITLE
Adding support for the RP2350 SoC, Raspberry Pi Pico 2 board, etc

### DISCUSSION THREAD

**Discussion by ajf58** [zephyrproject-rtos/zephyr#77329] [General]
This conversation began on Discord, and it was suggested that we start a discussion here.

I'd like to add support for the new RP2350 SoC from Raspberry Pi, along with initial board support for their Raspberry Pi Pico 2, which is one of the new boards based around this SoC.

Currently, I've got enough changes in place to build and run `samples/hello_world` on the Cortex-M33 core 0 on the Pico 2 board. I'm going to tidy up/create a set of branches that are suitable to share more widely. My objective is to get enough core functionality that initial SoC and board support can be merged into `main`. Non-goals for me (currently) are booting into the RISC-V Hazard3 cores, supporting some of the new peripherals/features, etc. I think they'll be a few things we'll want to discuss/agree, including what is a MVP. This is building on a lot of the excellent work of @yonsch , @soburi and many others of course, and I'd welcome your input.

### Dependencies

- https://github.com/zephyrproject-rtos/zephyr/pull/76986
- https://github.com/zephyrproject-rtos/hal_rpi_pico/pull/6

cc: @yonsch and @soburi

**Comment by soburi** [comment]
> Currently, I've got enough changes in place to build and run samples/hello_world on the Cortex-M33 core 0 on the Pico 2 board.

Please create a draft PR when you're ready.
Having actual code will make the discussion more concrete and helpful.

> My objective is to get enough core functionality that initial SoC and board support can be merged into main. Non-goals for me (currently) are booting into the RISC-V Hazard3 cores, supporting some of the new peripherals/features, etc. I think they'll be a few things we'll want to discuss/agree, including what is a MVP. This is building on a lot of the excellent work of @yonsch , @soburi and many others of course, and I'd welcome your input.

First of all, I think it would be good to create an initial version that supports peripherals that can be used in common with the Pico1.
The default settings should be highly compatible with the Pico1, and in the process, I think the excess and deficiencies in functionality will be sorted out and the direction will be clarified.

This will make it easier for a variety of people to work on development.
(It's not good to have one person monopolize an interesting development!)

**Reply by ajf58** [reply]
> Please create a draft PR when you're ready

Sure thing! https://github.com/zephyrproject-rtos/zephyr/pull/77368

**Comment by znmeb** [comment]
I have some boards (Pico 2, Pimoroni Pico Plus 2, SparkFun Pro Micro RP2350, iLabs Challenger+ RP2350) I can test on when this is ready. All but the Pico 2 have PSRAM!.

**Comment by dkouba-atym** [comment]
Has anyone gotten PSRAM working on the RP2350? I don't see anything in the available drivers that calls it out.

**Comment by znmeb** [comment]
I don't know about Zephyr, but the Raspberry Pi Pico SDK and Earle Philhower's Arduino Pico core both support PSRAM.
